### PR TITLE
E2E tests for `eth_createAccessList` and `eth_getBlockReceipts`

### PR DESCRIPTION
**Pull request by novosandara**
# Description

The PR implements e2e tests for `eth_createAccessList` and `eth_getBlockReceipts` JSON RPC functions.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

# Documentation update

# Additional comments

### CODE REVIEW

File: jsonrpc/types.go [hunk 1]
```diff
@@ -245,7 +245,7 @@ func toHeader(h *types.Header) *header {
 	return res
 }
 
-type receipt struct {
+type Receipt struct {
```
Comment (goran-ethernal) [inline jsonrpc/types.go]: Do not expose this struct, it is only used to serialize data when sending it as a request result. Just use the `ethgo.Receipt` when deserializing in `EthClient`.

File: e2e-polybft/e2e/jsonrpc_test.go [hunk 2]
```diff
@@ -302,11 +301,73 @@ func TestE2E_JsonRPC(t *testing.T) {
...
+		require.True(t, deployTxn.Succeed())
+		receipt := deployTxn.Receipt()
+
+		target := types.Address(receipt.ContractAddress)
+		input := contractsapi.TestSimple.Abi.GetMethod("getValue").ID()
+
+		gasPrice, err := newEthClient.GasPrice()
+		require.NoError(t, err)
+
+		// setValueFn := contractsapi.TestSimple.Abi.GetMethod("setValue")
```
Comment (goran-ethernal) [inline e2e-polybft/e2e/jsonrpc_test.go]: Remove this comment if not needed.

**Review by goran-ethernal — APPROVED** [review]

File: e2e-polybft/e2e/jsonrpc_test.go (hunk 2 above)
Comment (Stefan-Ethernal) [reply e2e-polybft/e2e/jsonrpc_test.go]: The idea was to send a transaction that would set the value to some non-zero value, and at the end to assert that access list result contain storage key for the `value` field as well.

Something like this, but for Input, provided the result of `setValueFn.Encode(...)` function

```go
txn := types.NewTx(
			types.NewLegacyTx(
				types.WithFrom(newAccountAddr), // adapt
				types.WithTo(&target),
				types.Input(...), //setValueEncoded byte array,
			))
	
		signedTxn, err := crypto.NewLondonSigner(chainID.Uint64()).SignTx(txn, newAccountKey) // use the proper account key
		require.NoError(t, err)
		data := signedTxn.MarshalRLPTo(nil)
		hash, err := newEthClient.SendRawTransaction(data)
		require.NoError(t, err)
		require.NotEqual(t, types.ZeroHash, hash)
```

File: jsonrpc/types.go [hunk 3]
```diff
@@ -564,15 +564,15 @@ func (s StateOverride) MarshalJSON() ([]byte, error) {
...
-
+	From       types.Address  // the sender of the 'transaction'
+	To         *types.Address // the destination contract (nil for contract creation)
+	Gas        uint64         // if 0, the call executes with near-infinite gas
+	GasPrice   *big.Int       // wei <-> gas exchange ratio
+	GasFeeCap  *big.Int       // EIP-1559 fee cap per gas
+	GasTipCap  *big.Int       // EIP-1559 tip per gas
+	Value      *big.Int       // amount of wei sent along with the call
+	Data       []byte         // input data, usually an ABI-encoded contract method invocation
+	Type       uint64
```
Comment (goran-ethernal) [inline jsonrpc/types.go:575]: Why do we need type here? I don't see it being used anywhere.

**Review by Stefan-Ethernal — APPROVED** [review]

**Pull request merged**